views/js/qtiCreator/widgets/interactions/textEntryInteraction/states/Correct.js

```diff
@@ -86,19 +86,25 @@ define([
             const convertedValue = converter.convert(value.trim());
             switch (attributes.baseType) {
                 case 'integer':
+                    if (value === '') {
+                        break;
+                    }
                     value = locale.parseInt(convertedValue, numericBase);
-                    responseValue = isNaN(value) ? '' : value;
+                    responseValue = isNaN(value) ? '' : value
                     // check for parsing and integer
-                    if (responseValue === '' || !/^[+-]?[0-9]+(e-?\d*)?$/.test(convertedValue)) {
+                    if (!/^[+-]?[0-9]+(e-?\d*)?$/.test(convertedValue)) {
```

Comment:
I know its not your regex but here we have ability to but scientific notation using `e` which isn't integer number. For example  put `123e5` , will it break save or other part of code?

Reply:
It is not a part of this ticket to change this regex.